engine/src/main/java/org/camunda/bpm/engine/impl/dmn/cmd/DeleteHistoricDecisionInstancesBatchCmd.java

```diff
@@ -53,44 +50,39 @@ public DeleteHistoricDecisionInstancesBatchCmd(List<String> ids,
 
   @Override
   public Batch execute(CommandContext commandContext) {
-    Collection<String> collectedInstanceIds = collectHistoricDecisionInstanceIds();
-    ensureNotEmpty(BadUserRequestException.class,
-        "historicDecisionInstanceIds", collectedInstanceIds);
+    BatchElementConfiguration elementConfiguration = collectHistoricDecisionInstanceIds(commandContext);
+    ensureNotEmpty(BadUserRequestException.class, "historicDecisionInstanceIds", elementConfiguration.getIds());
 
     return new BatchBuilder(commandContext)
         .type(Batch.TYPE_HISTORIC_DECISION_INSTANCE_DELETION)
-        .config(getConfiguration(collectedInstanceIds))
+        .config(getConfiguration(elementConfiguration))
         .permission(BatchPermissions.CREATE_BATCH_DELETE_DECISION_INSTANCES)
         .operationLogHandler(this::writeUserOperationLog)
         .build();
   }
 
-  protected List<String> collectHistoricDecisionInstanceIds() {
+  protected BatchElementConfiguration collectHistoricDecisionInstanceIds(CommandContext commandContext) {
 
-    Set<String> collectedDecisionInstanceIds = new HashSet<String>();
+    BatchElementConfiguration elementConfiguration = new BatchElementConfiguration();
 
-    List<String> decisionInstanceIds = getHistoricDecisionInstanceIds();
-    if (decisionInstanceIds != null) {
-      collectedDecisionInstanceIds.addAll(decisionInstanceIds);
+    if (!CollectionUtil.isEmpty(historicDecisionInstanceIds)) {
+      HistoricDecisionInstanceQueryImpl query = new HistoricDecisionInstanceQueryImpl();
+      query.decisionInstanceIdIn(historicDecisionInstanceIds.toArray(new String[historicDecisionInstanceIds.size()]));
```

Comment:
```suggestion
      query.decisionInstanceIdIn(historicDecisionInstanceIds.toArray(new String[0]));
```

Reply:
Thanks for that learning (I just read https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_empty_array_instantiation)!